spec/lib/cms/audit/pages_spec.rb

```diff
@@ -0,0 +1,111 @@
+describe Cms::Audit::Pages do
+  let(:role) { :user }
+  let(:current_user) { create(:user, role: Comfy::Cms::User.roles[role]) }
+  let(:state) { 'published' }
+  let(:en_site) { create(:site) }
+  let(:cy_site) { create(:site, label: 'cy') }
+  let(:page_with_tags) { create(:page_with_multiple_tags, site: en_site, state: state) }
+  let(:page_with_meta) { create(:article_with_metadata, site: cy_site, state: state) }
+  let!(:revision) do
+    create(
+      :revision_with_event,
+      record: page_with_tags,
+      user_id: current_user.id,
+      user_name: current_user.name
+    )
+  end
+  let!(:revision_2) do
+    create(
+      :revision_with_event,
+      record: page_with_meta,
+      user_id: current_user.id,
+      user_name: current_user.name
+    )
+  end
+  let!(:category) { create(:category, site: en_site) }
+  let!(:categorization) do
+    create(
+      :categorization,
+      categorized_id: page_with_tags.id,
+      category: category,
+      categorized_type: 'Comfy::Cms::Page'
+    )
+  end
+
+  before do
+    Timecop.freeze(Time.zone.local(1990))
+  end
+
+  describe 'self.generate' do
+    subject { described_class.generate_report(file: temp_file) }
+
+    let(:temp_file) { Tempfile.new('temp_file.csv') }
+
+    it 'generates a csv file and returns the count of pages added' do
+      expect(subject).to eq 2
+    end
+
+    context 'csv file content' do
```

Comment:
nitpick: It looks like this, being a specialisation of what to check and not a "when this happens.../with this setup...", suits better a `describe` block than a `context` (even if they're actually the same)